includes/payment-methods/class-wc-stripe-express-checkout-helper.php

```diff
@@ -571,92 +571,72 @@ public function is_page_supported() {
-		if (
-			WC_Stripe_Helper::has_cart_or_checkout_on_current_page()
-			&& ! $this->allowed_items_in_cart()
-		) {
-			WC_Stripe_Logger::log( 'Some items in cart are not compatible with Stripe Express Checkout. ' );
-			return false;
-		}
-
-		// Don't show on cart if disabled.
-		if ( is_cart() && ! $this->should_show_ece_on_cart_page() ) {
-			WC_Stripe_Logger::log( 'Stripe Express Checkout buttons display on cart is disabled. ' );
-			return false;
-		}
-
-		// Don't show on checkout if disabled.
-		if ( is_checkout() && ! $this->should_show_ece_on_checkout_page() ) {
-			WC_Stripe_Logger::log( 'Stripe Express Checkout buttons display on checkout is disabled. ' );
-			return false;
-		}
-
-		// Don't show if product page ECE is disabled.
-		if ( $this->is_product() && ! $this->should_show_ece_on_product_pages() ) {
-			WC_Stripe_Logger::log( 'Stripe Express Checkout buttons display on product pages is disabled. ' );
-			return false;
-		}
-
-		// Don't show if product on current page is not supported.
-		if ( $this->is_product() && ! $this->is_product_supported( $this->get_product() ) ) {
-			WC_Stripe_Logger::log( 'Product is not supported by Stripe Express Checkout. Product ID: ' . $this->get_product()->get_id() );
-			return false;
-		}
-
-		// Don't show if the total price is 0.
-		// ToDo: support free trials. Free trials should be supported if the product does not require shipping.
-		if ( ( ! ( $this->is_pay_for_order_page() || $this->is_product() ) && isset( WC()->cart ) && 0.0 === (float) WC()->cart->get_total( false ) )
-			|| ( $this->is_product() && 0.0 === (float) $this->get_product()->get_price() )
-		) {
-			WC_Stripe_Logger::log( 'Stripe Express Checkout does not support free products.' );
-			return false;
-		}
-
-		if ( $this->is_product() && in_array( $this->get_product()->get_type(), [ 'variable', 'variable-subscription' ], true ) ) {
-			$stock_availability = array_column( $this->get_product()->get_available_variations(), 'is_in_stock' );
-			// Don't show if all product variations are out-of-stock.
-			if ( ! in_array( true, $stock_availability, true ) ) {
-				WC_Stripe_Logger::log( 'Stripe Express Checkout is hidden due product variations being out of stock. Product ID: ' . $this->get_product()->get_id() );
-				return false;
-			}
-		}
-
-		// Hide if cart/product doesn't require shipping and tax is based on billing or shipping address.
-		if (
-			! $this->is_pay_for_order_page() &&
-			(
-				( is_product() && ! $this->product_needs_shipping( $this->get_product() ) ) ||
-				( ( is_cart() || is_checkout() ) && ( ! WC()->cart || ! WC()->cart->needs_shipping() ) )
-			) &&
-			wc_tax_enabled() &&
-			in_array( get_option( 'woocommerce_tax_based_on' ), [ 'billing', 'shipping' ], true )
-		) {
-			WC_Stripe_Logger::log( 'Stripe Express Checkout is hidden due product/cart not requiring shipping and tax being based on customer\'s billing or shipping address.' );
-			return false;
+		$should_show_button = false;
```

Comment:
To improve readability, I replaced the multiple IF conditions with a switch clause. To reduce the length of some lines, I created new variables, which are below.